kbase-extension/static/kbase/js/widgets/function_output/kbaseSampleSet.js

```diff
@@ -0,0 +1,228 @@
+                                    for (const meta_key in sample[0]['node_tree'][i]['meta_controlled']){
+                                        samp_data[meta_key] = self.unpack_metadata_to_string(
+                                            sample[0]['node_tree'][i]['meta_controlled'][meta_key]
+                                        );
+                                    }
+                                    for (const meta_key in sample[0]['node_tree'][i]['meta_user']){
+                                        samp_data[meta_key] = self.unpack_metadata_to_string(
+                                            sample[0]['node_tree'][i]['meta_user'][meta_key]
+                                        );
+                                    }
+                                }
+                                sample_data.push(samp_data)
+                            })
+                        )
+                        var row = [sample_info['name'], sample_info['id']];
+                        rows.push(row);
+                    });
+                    return Promise.all(sample_queries).then(function(){
+                        for (let j = 0; j < rows.length; j++){
+                            var row = rows[j];
+                            for (const meta_idx in self.metadata_headers){
+                                var meta_header = self.metadata_headers[meta_idx]
+                                console.log('in promise all')
+                                console.log(sample_data[j])
+                                console.log(meta_header)
+                                console.log(meta_header.id)
+                                console.log(meta_header.text)
+                                console.log(sample_data[j][meta_header.text])
+                                var row_str = self.options.default_blank_value;
+                                if (meta_header.text in sample_data[j]){
+                                    row_str = sample_data[j][meta_header.text]
+                                }
+                                row.push(row_str);
+                            }
+                            console.log(self.metadata_headers)
+                            rows[j] = row
+                        }
+                        return {
+                            rows: rows,
+                            start: pageNum * self.options.pageLimit,
+                            query: query,
+                            total: self.ss_obj_data['samples'].length
+                        }
+                    })
+                },
+                style: {'margin-top': '5px'}
+            });
+            return $content;
+        },
+
+        unpack_metadata_to_string: function(metadata){
+            var metastr = metadata['value'];
+            if ('units' in metadata){
+                metastr =  String(metadata['value']) + " " + String(metadata['units']);
```

Comment:
You can just add the string units to metastr if units are there. `metastr += ' ' + String(metadata['units']);`